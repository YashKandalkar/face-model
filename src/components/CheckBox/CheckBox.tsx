import React from "react";

interface Props {
  children?: React.ReactNode;
  className?: string;
  defaultChecked?: boolean;
  name?: string;
}

const CheckBox = (
  { children, className, defaultChecked, name }: Props,
  ref: React.Ref<HTMLInputElement>
) => {
  return (
    <label className={className}>
      {children}
      <input
        defaultChecked={defaultChecked}
        name={name}
        ref={ref}
        type="checkbox"
      />
      <span />
      <style jsx>{`
        label {
          color: inherit;
          cursor: pointer;
          display: block;
          padding-left: 35px;
          position: relative;
          user-select: none;
          width: max-content;
          -webkit-tap-highlight-color: transparent;
        }
        input {
          cursor: pointer;
          height: 0;
          opacity: 0;
          position: absolute;
          width: 0;
        }
        span {
          background-color: #454545;
          border-radius: 2px;
          position: absolute;
          height: 20px;
          width: 20px;
          left: 0;
          top: 0;
        }
        span:after {
          content: "";
          position: absolute;
          display: none;
        }
        input:checked ~ span:after {
          display: block;
        }
        span:after {
          left: 7px;
          top: 2px;
          width: 6px;
          height: 12px;
          border: solid #fff;
          border-width: 0 1px 1px 0;
          transform: rotate(45deg);
        }
      `}</style>
    </label>
  );
};

export default React.forwardRef(CheckBox);
