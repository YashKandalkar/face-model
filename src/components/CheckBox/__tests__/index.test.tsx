import React from "react";
import { mount } from "enzyme";
import CheckBox from "../index";

describe("CheckBox", () => {
  it("should be defined", () => {
    expect(CheckBox).toBeDefined();
  });

  it("should be rendered", () => {
    const wrapper = mount(<CheckBox />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render children", () => {
    const wrapper = mount(<CheckBox>CheckBox</CheckBox>);
    expect(wrapper.text()).toContain("CheckBox");
  });

  it("should render different text", () => {
    const wrapper = mount(<CheckBox>Text</CheckBox>);
    expect(wrapper.text()).toContain("Text");

    wrapper.setProps({
      children: <span>文本</span>,
    });
    expect(wrapper.text()).toContain("文本");
  });

  it("should forward ref by default", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<CheckBox ref={ref} />);
    expect(ref.current).not.toBeNull();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should get checked value", () => {
    const wrapper = mount(<CheckBox defaultChecked={true}>Checked</CheckBox>);
    const input = wrapper.find("input").getDOMNode();
    expect((input as HTMLInputElement).checked).toBeTruthy();
  });

  it("should change checked value", () => {
    const ref = React.createRef<HTMLInputElement>();
    const wrapper = mount(<CheckBox ref={ref} />);
    expect(ref.current?.checked).toBe(false);
    wrapper.find("input").getDOMNode<HTMLInputElement>().checked = true;
    wrapper.find("input").simulate("change");
    expect(ref.current?.checked).toBe(true);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should match snapshot", () => {
    expect(<CheckBox>CheckBox</CheckBox>).toMatchSnapshot();
    expect(
      <CheckBox defaultChecked={true}>CheckBox</CheckBox>
    ).toMatchSnapshot();
  });
});
