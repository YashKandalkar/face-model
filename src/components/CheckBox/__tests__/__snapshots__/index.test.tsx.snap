// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should match snapshot 1`] = `
<ForwardRef(CheckBox)>
  CheckBox
</ForwardRef(CheckBox)>
`;

exports[`CheckBox should match snapshot 2`] = `
<ForwardRef(CheckBox)
  defaultChecked={true}
>
  CheckBox
</ForwardRef(CheckBox)>
`;
